### PR TITLE
[model] support FireRedLID

### PR DESCRIPTION
**Pull request by PatchouliTIS**
## Purpose

Hi @DarkLight1337  ,  this PR adds support for the FireRedLID model ([PatchyTisa/FireRedLID-vllm](https://huggingface.co/PatchyTisa/FireRedLID-vllm)). FireRedLID is the spoken language identification module from the [FireRedASR2S](https://github.com/FireRedTeam/FireRedASR2S) system, capable of identifying 100+ languages and 20+ Chinese dialects. Would you kindly take a look?

Architecture: ConformerEncoder + TransformerDecoder (6-layer cross-attention), following the Whisper-style encoder-decoder pattern.

Server:

```bash
vllm serve PatchyTisa/FireRedLID-vllm -tp=1 --dtype=float32
```
Also supports  `-tp=2 --dtype=bfloat16`.

Client:

```bash
python examples/online_serving/openai_lid_client.py --audio_paths /path/to/audio_en.wav /path/to/audio_zh.wav
```

Notes:

FireRedLID shares the same Conformer encoder architecture as FireRedASR2 (including RelPosMultiHeadAttention). Same as the FireRedASR2 PR, I still haven't found a better way to replace it with MMEncoderAttention.
The model re-uses the OpenAI-compatible /v1/audio/transcriptions endpoint — the "transcription" it returns is actually the language label.

## Test Result

---
<details>
<summary> Essential Elements of an Effective PR Description Checklist </summary>

- [ ] The purpose of the PR, such as "Fix some issue (link existing issues this PR will resolve)".
- [ ] The test plan, such as providing test command.
- [ ] The test results, such as pasting the results comparison before and after, or e2e results
- [ ] (Optional) The necessary documentation update, such as updating `supported_models.md` and `examples` for a new model.
- [ ] (Optional) Release notes update. If your change is user facing, please update the release notes draft in the [Google Doc](https://docs.google.com/document/d/1YyVqrgX4gHTtrstbq8oWUImOyPCKSGnJ7xtTpmXzlRs/edit?tab=t.0).
</details>

### CODE REVIEW

**Review by DarkLight1337 — COMMENTED** [review]
Similar to https://github.com/vllm-project/vllm/pull/37446, can you factor out the common models with `fireredasr2.py` into a separate file?

File: examples/online_serving/openai_lid_client.py [hunk 1]
```diff
@@ -0,0 +1,193 @@
+# SPDX-License-Identifier: Apache-2.0
```
Comment (DarkLight1337) [inline examples/online_serving/openai_lid_client.py:1]: Can you add this model to the offline example as well?

File: docs/models/supported_models.md [hunk 2]
```diff
@@ -660,6 +660,7 @@ Speech2Text models trained specifically for Automatic Speech Recognition.
 | ------------ | ------ | ----------------- | -------------------- | ------------------------- |
 | `CohereAsrForConditionalGeneration` | Cohere-Transcribe | `CohereLabs/cohere-transcribe-03-2026` | | |
 | `FireRedASR2ForConditionalGeneration` | FireRedASR2 | `allendou/FireRedASR2-LLM-vllm`, etc. | | |
+| `FireRedLIDForConditionalGeneration` | FireRedLID | `allendou/FireRedLID-vllm`, etc. | | |
```
Comment (DarkLight1337) [inline docs/models/supported_models.md]:
```suggestion
| `FireRedLIDForConditionalGeneration` | FireRedLID | `PatchyTisa/FireRedLID-vllm`, etc. | | |
```

File: vllm/model_executor/models/conformer_encoder.py [hunk 3]
```diff
@@ -0,0 +1,357 @@
...
+This module factors out the common building blocks to avoid duplication.
+"""
+
+import torch
+import torch.nn.functional as F
+from torch import nn
+
+from vllm.model_executor.layers.linear import ReplicatedLinear
+
+# ===========================================================================
```
Comment (DarkLight1337) [inline vllm/model_executor/models/conformer_encoder.py]: Please remove these AI comments

File: vllm/multimodal/processing/processor.py [hunk 4]
```diff
@@ -1747,6 +1747,15 @@ def _get_enc_dec_inputs(
             decoder_prompt_text = None
             decoder_prompt_ids = decoder_prompt_raw
 
+        # Ensure decoder_start_token_id is prepended, consistent with
```
Comment (DarkLight1337) [inline vllm/multimodal/processing/processor.py]: This shouldn't be necessary because we always call `_prepare_decoder_input_ids_for_generation`

Comment (PatchouliTIS) [reply vllm/multimodal/processing/processor.py]: Removed codes from `processor.py`

File: vllm/model_executor/models/conformer_encoder.py (hunk 3 above)
Comment (PatchouliTIS) [reply vllm/model_executor/models/conformer_encoder.py]: done

File: examples/online_serving/openai_lid_client.py (hunk 1 above)
Comment (PatchouliTIS) [reply examples/online_serving/openai_lid_client.py:1]: Added offline examples in `examples/offline_inference/encoder_decoder_multimodal.py` and `examples/offline_inference/audio_language.py`

File: examples/offline_inference/encoder_decoder_multimodal.py [hunk 5]
```diff
@@ -55,7 +55,43 @@ def run_whisper():
     )
 
 
+def run_fireredlid():
```
Comment (DarkLight1337) [inline examples/offline_inference/encoder_decoder_multimodal.py:98]: To help check the correctness can you add an example for FireRedASR2 as well? So we can compare the results before and after this PR

Comment (PatchouliTIS) [reply examples/offline_inference/encoder_decoder_multimodal.py:98]: Add offline demo for FireRedASR2S, got output in local test:
```
Decoder prompt: '<|im_start|>user\n<|AUDIO|>请转写音频为文字<|im_end|>\n<|im_start|>assistant\n', Generated text: 'thefirstwordsispokeintheoriginalcornographlittlepieceofpracticalpoetrymaryhadalittlelamitsmoothwithwhitesnowandeverywherethatmarywentthelambwouldshootago'
Decoder prompt: '<|im_start|>user\n<|AUDIO|>请转写音频为文字<|im_end|>\n<|im_start|>assistant\n', Generated text: "and the old one pitch on the way to edgar martinez swung on the line up the line for a base hit here comes joy here is junior to third base they're gonna wave him in the throw to the plate will be late the mariners are going to play for the american league championship i don't believe it"
Duration: 0.5372672080993652
RPS: 3.722542470208062
```

same as old version.

**Review by DarkLight1337 — APPROVED** [review]
Thanks, LGTM
